### PR TITLE
Make default terminate API configurable to support graceful termination

### LINKED ISSUE #506: ExitHandler not triggered on pipeline run cancellation
/kind bug

**What steps did you take and what happened:**
When cancelling a pipeline run, operations added with `ExitHandler` are not triggered.

Operations added with `ExitHandler` are mapped to the Tekton final tasks (`finally`).
Tekton final tasks are skipped when Tekton `PipelineRun` is [cancelled](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#cancelling-a-pipelinerun).

**What did you expect to happen:**
A graceful pipeline run cancellation should guarantee that operations added with `ExitHandler` are triggered on run stop.

**Additional information:**
Related Argo issue: https://github.com/argoproj/argo-workflows/issues/3155

**Environment:**

* Python Version (use `python --version`): 3.7
* SDK Version: 0.7.0
* Tekton Version (use `tkn version`): 0.22
* Kubernetes Version (use `kubectl version`): 1.18
* OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #506 

**Description of your changes:**
Make the terminate API status to be configurable for the admin. This way the admin can choose whether to let the finally tasks to run or not.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`): 0.25.0
* Kubernetes Version (use `kubectl version`): 1.18
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Tomcli** [comment]
/hold

@rafalbigaj Do you think this can satisfy your requirements? Let us know how you would like us to implement this feature into KFP. Thanks.

**Comment by Tomcli** [comment]
cc @Udiknedormin

File: backend/src/apiserver/resource/resource_manager.go [hunk 1]
```diff
@@ -519,7 +519,7 @@ func (r *ResourceManager) ListJobs(filterContext *common.FilterContext,
 func TerminateWorkflow(wfClient workflowclient.PipelineRunInterface, name string) error {
 	patchObj := map[string]interface{}{
 		"spec": map[string]interface{}{
-			"status": "Cancelled",
+			"status": common.GetTerminateStatus(),
```
Comment (pugangxa) [inline backend/src/apiserver/resource/resource_manager.go:522]: This is the default behavior, do we support user to overwrite it by explicitly specifying the parameter?

Comment (Tomcli) [reply backend/src/apiserver/resource/resource_manager.go:522]: This PR allows the admin to configure this status, not users. If we want to allow users to configure this, we need to have some new annotations/dsl to indicate this new status.

**Comment by Tomcli** [comment]
So after the meetings today, we agree not to expose this feature on the API level since we don't want users to control the terminate behavior. So this feature should be good to go.
/unhold

**Comment by Tomcli** [comment]
@ckadner @pugangxa @ScrapCodes maybe one of you can review this PR? thanks.

**Comment by pugangxa** [comment]
/lgtm
/approve
It's straightforward to me, as long as tekton can handle it as expected, thanks.

**Pull request merged**